mmv1/products/networkservices/TlsRoute.yaml

```diff
@@ -0,0 +1,159 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Resource
+name: 'TlsRoute'
+base_url: 'projects/{{project}}/locations/global/tlsRoutes'
+create_url: 'projects/{{project}}/locations/global/tlsRoutes?tlsRouteId={{name}}'
+self_link: 'projects/{{project}}/locations/global/tlsRoutes/{{name}}'
+min_version: beta
+update_verb: :PATCH
+update_mask: true
+description: |
+  TlsRoute defines how traffic should be routed based on SNI and other matching L3 attributes.
+references: !ruby/object:Api::Resource::ReferenceLinks
+  api: 'https://cloud.google.com/traffic-director/docs/reference/network-services/rest/v1beta1/projects.locations.tlsRoutes'
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+    timeouts: !ruby/object:Api::Timeouts
+      insert_minutes: 30
+      update_minutes: 30
+      delete_minutes: 30
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+autogen_async: true
+import_format: ['projects/{{project}}/locations/global/tlsRoutes/{{name}}']
+examples:
+  - !ruby/object:Provider::Terraform::Examples
+    min_version: beta
+    name: 'network_services_tls_route_basic'
+    primary_resource_id: 'default'
+    vars:
+      resource_name: 'my-tls-route'
+      backend_service_name: "my-backend-service"
+      http_health_check_name: "backend-service-health-check"
+  - !ruby/object:Provider::Terraform::Examples
+    min_version: beta
+    name: 'network_services_tls_route_mesh_basic'
+    primary_resource_id: 'default'
+    vars:
+      resource_name: 'my-tls-route'
+      backend_service_name: "my-backend-service"
+      http_health_check_name: "backend-service-health-check"
+  - !ruby/object:Provider::Terraform::Examples
+    min_version: beta
+    name: 'network_services_tls_route_gateway_basic'
+    primary_resource_id: 'default'
+    vars:
+      resource_name: 'my-tls-route'
+      backend_service_name: "my-backend-service"
+      http_health_check_name: "backend-service-health-check"
+parameters:
+  - !ruby/object:Api::Type::String
+    name: 'name'
+    required: true
+    url_param_only: true
+    immutable: true
+    description: |
+      Name of the TlsRoute resource.
+properties:
+  - !ruby/object:Api::Type::String
+    name: 'selfLink'
+    description: |
+      Server-defined URL of this resource.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'createTime'
+    description: |
+      Time the TlsRoute was created in UTC.
+    output: true
+  - !ruby/object:Api::Type::Time
+    name: 'updateTime'
```

Comment:
Are these fields going to be used? create_time and update_time aren't normally needed fields in Terraform and could be removed if we don't expect a use case. update_time in particular is a transient value that could not be relied on in code.

Reply:
They have been marked as output only fields.

Reply:
Even as output only fields their inclusion is generally not needed or useful if there is no special reason to be including them

Reply:
Even as output only fields, their inclusion is generally not useful or needed in the terraform state.

Reply:
I would keep this for the sake of consistency with the majority of the resources.